### ISSUE TITLE
sklearn support

### ISSUE THREAD

**Issue by methodds** [bmabey/pyLDAvis#37]
Hi, first of all thanks a lot for the R port of this package :). Is there any chance for that you add model support for sklearn? They just released LDA in their kit:
http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.LatentDirichletAllocation.html#sklearn.decomposition.LatentDirichletAllocation

**Comment by bmabey** [comment]
Yeah, I'd merge in a PR with support. I probably won't get around to adding support myself anytime too soon.

-Ben

**Comment by methodds** [comment]
I'm afraid my programming is not good enough to implement this, although I would like to try it. How would you approach this problem? Some short hints might be very useful for me :)

**Comment by bmabey** [comment]
I would recommend to start coding it in a notebook so you can iterate fast on it and get visual checks of what you did works. The goal is the take the parameters learned/fit by sklearn's LDA model and transform them into the shape and format that pyLDAvis' [prepare](https://pyldavis.readthedocs.org/en/latest/modules/API.html#pyLDAvis.prepare) function expect. The function is well documented and so hopefully that is all you need to get started. I'm not familiar with sklearn's LDA impl. so I can't comment on how to start their aside from looking at the docs and code. Let me know if you have any questions. If you push a WIP notebook as a gist I could take a look if you get stuck.

**Comment by napjon** [comment]
Please give feedback.
https://gist.github.com/napjon/ef842b41e0048d132607

Strange thing is that sklearn LDA return doc_topic_dists that isn't sum up to 1 for each row.  So I have to normalized it.

**Comment by napjon** [comment]
Hi @bmabey, can you check the function? it should work with LDA module from https://github.com/ariddell/lda as well. Can you suggest where do I put the function if I want to pull request?

**Comment by bmabey** [comment]
Thanks @napjon. The implementation looks good on first glance. I would suggest mirroring the structure of [pyLDAvis.gensim](https://github.com/bmabey/pyLDAvis/blob/master/pyLDAvis/gensim.py). Having the `_extract_data` function separate can be helpful for debugging and then you can rename the `from_sklearn` as `prepare` to match convention.

Also, could you create a notebook showing it being used with sklearn and the other LDA library?

Thanks!

**Comment by bmabey** [comment]
To be clear, I am suggesting that you create `pyLDAvis.sklearn` with two functions: `_extract_data` and `prepare`.

**Comment by napjon** [comment]
Got it, let me try to create it.
